# Ebay Car Sales Data

Project to explore Ebay Car Sales Data. The data contains car details. Aim of the project is to clean the data and analyze the included used car listings.

In [1]:
# Imports, read csv
import pandas as pd
import numpy as np

autos = pd.read_csv('autos.csv',encoding = 'Latin-1')

In [2]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [3]:
# Info and head
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

# Observations
There are 50000 cars in dataframe. vehicletype, fueltype, gearbox, model, notRepairedDamage columns have a few null rows. Other columns are not null.
There are 20 columns in dataframe.

In [4]:
# Print existing column names
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [5]:
# Edit column names and rename few to snake case
autos = autos.rename(columns = {"yearOfRegistration" : "registration_year",
                       "monthOfRegistration":"registration_month",
                       "notRepairedDamage":"unrepaired_damage",
                       "dateCreated":"ad_created"})
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  registration_year  \
0  privat   Angebot  $5,000  control         bus               2004   
1  privat   Angebot  $8,500  control   limousine               1997   
2  privat   Angebot  $8,990     test   limousine               2009   
3  privat   Angebot  $4,350  control  kleinwagen               2007   
4  privat   Angebot  $1,350     test       kombi               2003   

     gearbox  powerPS   model   odometer  registration_month fuelType  \
0    manuell      158  andere  150,000km                   3      lpg   
1  automatik      286     7er  150,000km                   6   benzin   
2    manuell      102    golf   70,000km                   7   benzin   
3  automatik       71  fortwo   70,000km                   6   benzin   
4    manuell        0   focus  150,000km                   7   benzin   

        brand unrepaired_damage           ad_created  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [6]:
# Basic data exploration
autos.describe()

registration_year       powerPS  registration_month  nrOfPictures  \
count       50000.000000  50000.000000        50000.000000       50000.0   
mean         2005.073280    116.355920            5.723360           0.0   
std           105.712813    209.216627            3.711984           0.0   
min          1000.000000      0.000000            0.000000           0.0   
25%          1999.000000     70.000000            3.000000           0.0   
50%          2003.000000    105.000000            6.000000           0.0   
75%          2008.000000    150.000000            9.000000           0.0   
max          9999.000000  17700.000000           12.000000           0.0   

         postalCode  
count  50000.000000  
mean   50813.627300  
std    25779.747957  
min     1067.000000  
25%    30451.000000  
50%    49577.000000  
75%    71540.000000  
max    99998.000000

In [7]:
# Describe including non-numeric columns (categorical)
autos.describe(include = 'all')

dateCrawled         name  seller offerType  price abtest  \
count                 50000        50000   50000     50000  50000  50000   
unique                48213        38754       2         2   2357      2   
top     2016-03-12 16:06:22  Ford_Fiesta  privat   Angebot     $0   test   
freq                      3           78   49999     49999   1421  25756   
mean                    NaN          NaN     NaN       NaN    NaN    NaN   
std                     NaN          NaN     NaN       NaN    NaN    NaN   
min                     NaN          NaN     NaN       NaN    NaN    NaN   
25%                     NaN          NaN     NaN       NaN    NaN    NaN   
50%                     NaN          NaN     NaN       NaN    NaN    NaN   
75%                     NaN          NaN     NaN       NaN    NaN    NaN   
max                     NaN          NaN     NaN       NaN    NaN    NaN   

       vehicleType  registration_year  gearbox       powerPS  model  \
count        44905       50000.000000    47320  50000.000000  47242   
unique           8                NaN        2           NaN    245   
top      limousine                NaN  manuell           NaN   golf   
freq         12859                NaN    36993           NaN   4024   
mean           NaN        2005.073280      NaN    116.355920    NaN   
std            NaN         105.712813      NaN    209.216627    NaN   
min            NaN        1000.000000      NaN      0.000000    NaN   
25%            NaN        1999.000000      NaN     70.000000    NaN   
50%            NaN        2003.000000      NaN    105.000000    NaN   
75%            NaN        2008.000000      NaN    150.000000    NaN   
max            NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuelType       brand unrepaired_damage  \
count       50000        50000.000000    45518       50000             40171   
unique         13                 NaN        7          40                 2   
top     150,000km                 NaN   benzin  volkswagen              nein   
freq        32424                 NaN    30107       10687             35232   
mean          NaN            5.723360      NaN         NaN               NaN   
std           NaN            3.711984      NaN         NaN               NaN   
min           NaN            0.000000      NaN         NaN               NaN   
25%           NaN            3.000000      NaN         NaN               NaN   
50%           NaN            6.000000      NaN         NaN               NaN   
75%           NaN            9.000000      NaN         NaN               NaN   
max           NaN           12.000000      NaN         NaN               NaN   

                 ad_created  nrOfPictures    postalCode             lastSeen  
count                 50000       50000.0  50000.000000                50000  
unique                   76           NaN           NaN                39481  
top     2016-04-03 00:00:00           NaN           NaN  2016-04-07 06:17:27  
freq                   1946           NaN           NaN                    8  
mean                    NaN           0.0  50813.627300                  NaN  
std                     NaN           0.0  25779.747957                  NaN  
min                     NaN           0.0   1067.000000                  NaN  
25%                     NaN           0.0  30451.000000                  NaN  
50%                     NaN           0.0  49577.000000                  NaN  
75%                     NaN           0.0  71540.000000                  NaN  
max                     NaN           0.0  99998.000000                  NaN

In [8]:
# Convert price and odometer columns to numeric
autos['odometer'] = autos['odometer'].str.replace("km","")
autos['odometer'] = autos['odometer'].str.replace(",","")
autos['odometer'] = autos['odometer'].astype(int)

autos['price'] = autos['price'].str.replace("$","")
autos['price'] = autos['price'].str.replace(",","")
autos['price'] = autos['price'].astype(int)

autos.tail()

dateCrawled                                               name  \
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType  price   abtest vehicleType  registration_year  \
49995  privat   Angebot  24900  control   limousine               2011   
49996  privat   Angebot   1980  control      cabrio               1996   
49997  privat   Angebot  13200     test      cabrio               2014   
49998  privat   Angebot  22900  control       kombi               2013   
49999  privat   Angebot   1250  control   limousine               1996   

         gearbox  powerPS   model  odometer  registration_month fuelType  \
49995  automatik      239      q5    100000                   1   diesel   
49996    manuell       75   astra    150000                   5   benzin   
49997  automatik       69     500      5000                  11   benzin   
49998    manuell      150      a3     40000                  11   diesel   
49999    manuell      101  vectra    150000                   1   benzin   

      brand unrepaired_damage           ad_created  nrOfPictures  postalCode  \
49995  audi              nein  2016-03-27 00:00:00             0       82131   
49996  opel              nein  2016-03-28 00:00:00             0       44807   
49997  fiat              nein  2016-04-02 00:00:00             0       73430   
49998  audi              nein  2016-03-08 00:00:00             0       35683   
49999  opel              nein  2016-03-13 00:00:00             0       45897   

                  lastSeen  
49995  2016-04-01 13:47:40  
49996  2016-04-02 14:18:02  
49997  2016-04-04 11:47:27  
49998  2016-04-05 16:45:07  
49999  2016-04-06 21:18:48

In [9]:
# Rename odometer column
autos = autos.rename(columns = {"odometer":"odometer_km"})
autos.head(2)

dateCrawled                                        name  seller  \
0  2016-03-26 17:47:46            Peugeot_807_160_NAVTECH_ON_BOARD  privat   
1  2016-04-04 13:38:56  BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik  privat   

  offerType  price   abtest vehicleType  registration_year    gearbox  \
0   Angebot   5000  control         bus               2004    manuell   
1   Angebot   8500  control   limousine               1997  automatik   

   powerPS   model  odometer_km  registration_month fuelType    brand  \
0      158  andere       150000                   3      lpg  peugeot   
1      286     7er       150000                   6   benzin      bmw   

  unrepaired_damage           ad_created  nrOfPictures  postalCode  \
0              nein  2016-03-26 00:00:00             0       79588   
1              nein  2016-04-04 00:00:00             0       71034   

              lastSeen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08

In [10]:
# Explore data for odometer_km column
print(autos['odometer_km'].unique().shape)
print(autos['odometer_km'].value_counts())
print(autos['odometer_km'].describe())

(13,)
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64
count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


# Odometer column analysis
We observe that there are only 13 unique values for odometer as they are probably rounded.

There is no major outlier, so we can keep all the rows.

In [11]:
# Explore data for price column
print("Unique prices:" + str(autos['price'].unique().shape))
print("Ascending prices:\n" + str(autos['price'].value_counts().
                                sort_index(ascending=True).head(10)))
print("Descending prices:\n" + str(autos['price'].value_counts().
                                 sort_index(ascending=False).head(20)))
print(autos['price'].describe())

Unique prices:(2357,)
Ascending prices:
0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
Name: price, dtype: int64
Descending prices:
99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
3890000     1
1300000     1
1234566     1
999999      2
999990      1
350000      1
345000      1
299000      1
295000      1
265000      1
259000      1
250000      1
220000      1
198000      1
197000      1
Name: price, dtype: int64
count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64


In [12]:
# Remove the outliers from price
autos = autos[autos['price'].between(1,350000)]
autos.describe()

price  registration_year       powerPS    odometer_km  \
count   48565.000000       48565.000000  48565.000000   48565.000000   
mean     5888.935591        2004.755421    117.197158  125770.101925   
std      9059.854754          88.643887    200.649618   39788.636804   
min         1.000000        1000.000000      0.000000    5000.000000   
25%      1200.000000        1999.000000     71.000000  125000.000000   
50%      3000.000000        2004.000000    107.000000  150000.000000   
75%      7490.000000        2008.000000    150.000000  150000.000000   
max    350000.000000        9999.000000  17700.000000  150000.000000   

       registration_month  nrOfPictures    postalCode  
count        48565.000000       48565.0  48565.000000  
mean             5.782251           0.0  50975.745207  
std              3.685595           0.0  25746.968398  
min              0.000000           0.0   1067.000000  
25%              3.000000           0.0  30657.000000  
50%              6.000000           0.0  49716.000000  
75%              9.000000           0.0  71665.000000  
max             12.000000           0.0  99998.000000

# Observations for Price column

There are few outliers in price column. 

1) Ascending: There can be some cars for sale at ebay with value of 1, so we will remove the 0 values

2) Descending: The cars sale price increases until 350,000 but then there is a big jump to 999,990. And there are some cars which are more expensive, and these are outliers, and do not seem to be legitimate for Ebay.

In [13]:
# Exploring the date columns
print(autos['dateCrawled'].str[:10].head())
print(autos['ad_created'].str[:10].head())
print(autos['lastSeen'].str[:10].head())


0    2016-03-26
1    2016-04-04
2    2016-03-26
3    2016-03-12
4    2016-04-01
Name: dateCrawled, dtype: object
0    2016-03-26
1    2016-04-04
2    2016-03-26
3    2016-03-12
4    2016-04-01
Name: ad_created, dtype: object
0    2016-04-06
1    2016-04-06
2    2016-04-06
3    2016-03-15
4    2016-04-01
Name: lastSeen, dtype: object


In [14]:
# Sorted by date
(autos["dateCrawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_index()
        * 100)

2016-03-05    2.532688
2016-03-06    1.404304
2016-03-07    3.601359
2016-03-08    3.329558
2016-03-09    3.308967
2016-03-10    3.218367
2016-03-11    3.257490
2016-03-12    3.691959
2016-03-13    1.566972
2016-03-14    3.654896
2016-03-15    3.428395
2016-03-16    2.960980
2016-03-17    3.162772
2016-03-18    1.291053
2016-03-19    3.477813
2016-03-20    3.788737
2016-03-21    3.737259
2016-03-22    3.298672
2016-03-23    3.222485
2016-03-24    2.934212
2016-03-25    3.160712
2016-03-26    3.220426
2016-03-27    3.109235
2016-03-28    3.486050
2016-03-29    3.409863
2016-03-30    3.368681
2016-03-31    3.183363
2016-04-01    3.368681
2016-04-02    3.547823
2016-04-03    3.860805
2016-04-04    3.648718
2016-04-05    1.309585
2016-04-06    0.317101
2016-04-07    0.140019
Name: dateCrawled, dtype: float64

In [15]:
# Sorted by descending percentage
(autos["dateCrawled"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values(ascending=False)
        * 100)

2016-04-03    3.860805
2016-03-20    3.788737
2016-03-21    3.737259
2016-03-12    3.691959
2016-03-14    3.654896
2016-04-04    3.648718
2016-03-07    3.601359
2016-04-02    3.547823
2016-03-28    3.486050
2016-03-19    3.477813
2016-03-15    3.428395
2016-03-29    3.409863
2016-03-30    3.368681
2016-04-01    3.368681
2016-03-08    3.329558
2016-03-09    3.308967
2016-03-22    3.298672
2016-03-11    3.257490
2016-03-23    3.222485
2016-03-26    3.220426
2016-03-10    3.218367
2016-03-31    3.183363
2016-03-17    3.162772
2016-03-25    3.160712
2016-03-27    3.109235
2016-03-16    2.960980
2016-03-24    2.934212
2016-03-05    2.532688
2016-03-13    1.566972
2016-03-06    1.404304
2016-04-05    1.309585
2016-03-18    1.291053
2016-04-06    0.317101
2016-04-07    0.140019
Name: dateCrawled, dtype: float64

In [16]:
# Sorted by descending percentage
(autos["ad_created"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values(ascending=False)
        * 100)

2016-04-03    3.885514
2016-03-20    3.794914
2016-03-21    3.757850
2016-04-04    3.685782
2016-03-12    3.675486
2016-03-14    3.518995
2016-04-02    3.514877
2016-03-28    3.498404
2016-03-07    3.473695
2016-03-29    3.403686
2016-03-15    3.401627
2016-04-01    3.368681
2016-03-19    3.368681
2016-03-30    3.350149
2016-03-08    3.331617
2016-03-09    3.315145
2016-03-11    3.290435
2016-03-22    3.280140
2016-03-26    3.226604
2016-03-23    3.206013
2016-03-10    3.189540
2016-03-31    3.187481
2016-03-25    3.175126
2016-03-17    3.127767
2016-03-27    3.098940
2016-03-16    3.012458
2016-03-24    2.928035
2016-03-05    2.289715
2016-03-13    1.700813
2016-03-06    1.531967
                ...   
2016-02-05    0.004118
2016-02-20    0.004118
2016-02-02    0.004118
2016-02-12    0.004118
2016-02-18    0.004118
2016-02-26    0.004118
2016-02-14    0.004118
2016-01-10    0.004118
2016-02-24    0.004118
2016-02-09    0.002059
2016-01-22    0.002059
2015-09-09    0.002059
2015-12-05 

In [17]:
# Sorted by descending percentage
(autos["lastSeen"]
        .str[:10]
        .value_counts(normalize=True, dropna=False)
        .sort_values(ascending=False)
        * 100)

2016-04-06    22.180583
2016-04-07    13.194688
2016-04-05    12.476063
2016-03-17     2.808607
2016-04-03     2.520334
2016-04-02     2.491506
2016-03-30     2.477093
2016-04-04     2.448265
2016-03-31     2.378256
2016-03-12     2.378256
2016-04-01     2.279419
2016-03-29     2.234119
2016-03-22     2.137342
2016-03-28     2.085864
2016-03-20     2.065273
2016-03-21     2.063214
2016-03-24     1.976732
2016-03-25     1.921137
2016-03-23     1.853186
2016-03-26     1.680222
2016-03-16     1.645218
2016-03-15     1.587563
2016-03-19     1.583445
2016-03-27     1.564913
2016-03-14     1.260167
2016-03-11     1.237517
2016-03-10     1.066612
2016-03-09     0.959539
2016-03-13     0.889529
2016-03-08     0.741275
2016-03-18     0.735097
2016-03-07     0.539483
2016-03-06     0.432410
2016-03-05     0.107073
Name: lastSeen, dtype: float64

# Observations of date columns
2016-04-06 has the highest percentage of last seen. The last 3 days April 5-7 have values > 10%.
Date Crawled and Ad created ranges from 0-4% with no date standing out.

In [18]:
# Registration Year
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

In [19]:
autos['registration_year'].value_counts().sort_index()

1000       1
1001       1
1111       1
1800       2
1910       5
1927       1
1929       1
1931       1
1934       2
1937       4
1938       1
1939       1
1941       2
1943       1
1948       1
1950       3
1951       2
1952       1
1953       1
1954       2
1955       2
1956       4
1957       2
1958       4
1959       6
1960      23
1961       6
1962       4
1963       8
1964      12
        ... 
2000    3156
2001    2636
2002    2486
2003    2699
2004    2703
2005    2936
2006    2670
2007    2277
2008    2215
2009    2085
2010    1589
2011    1623
2012    1310
2013     803
2014     663
2015     392
2016    1220
2017    1392
2018     470
2019       2
2800       1
4100       1
4500       1
4800       1
5000       4
5911       1
6200       1
8888       1
9000       1
9999       3
Name: registration_year, Length: 95, dtype: int64

# Registration Year
The correct years seem to be between 1910 and 2016, and there are few outliers of years, that are before cars were invented and also several years in the future.

In [20]:
# Remove the outliers from registration year
autos = autos[autos['registration_year'].between(1910,2016)]
autos.describe()

price  registration_year       powerPS    odometer_km  \
count   46681.000000       46681.000000  46681.000000   46681.000000   
mean     5977.716801        2002.910756    117.892933  125586.855466   
std      9177.909479           7.185103    184.922911   39852.528628   
min         1.000000        1910.000000      0.000000    5000.000000   
25%      1250.000000        1999.000000     75.000000  100000.000000   
50%      3100.000000        2003.000000    109.000000  150000.000000   
75%      7500.000000        2008.000000    150.000000  150000.000000   
max    350000.000000        2016.000000  17700.000000  150000.000000   

       registration_month  nrOfPictures    postalCode  
count        46681.000000       46681.0  46681.000000  
mean             5.827125           0.0  51097.434181  
std              3.670300           0.0  25755.387192  
min              0.000000           0.0   1067.000000  
25%              3.000000           0.0  30827.000000  
50%              6.000000           0.0  49828.000000  
75%              9.000000           0.0  71732.000000  
max             12.000000           0.0  99998.000000

In [21]:
# Value counts
autos['registration_year'].value_counts(normalize=True).sort_values(ascending=False)*100

2000    6.760781
2005    6.289497
1999    6.205951
2004    5.790364
2003    5.781796
2006    5.719672
2001    5.646837
2002    5.325507
1998    5.062017
2007    4.877788
2008    4.744971
2009    4.466485
1997    4.179431
2011    3.476789
2010    3.403954
1996    2.941239
2012    2.806281
1995    2.628478
2016    2.613483
2013    1.720186
2014    1.420278
1994    1.347443
1993    0.910435
2015    0.839742
1992    0.792614
1990    0.743343
1991    0.726206
1989    0.372743
1988    0.289197
1985    0.203509
          ...   
1977    0.047128
1976    0.044986
1969    0.040702
1975    0.038560
1965    0.036417
1964    0.025706
1963    0.017138
1961    0.012853
1959    0.012853
1910    0.010711
1956    0.008569
1958    0.008569
1937    0.008569
1962    0.008569
1950    0.006427
1954    0.004284
1941    0.004284
1951    0.004284
1934    0.004284
1957    0.004284
1955    0.004284
1939    0.002142
1927    0.002142
1948    0.002142
1938    0.002142
1931    0.002142
1929    0.002142
1943    0.0021

# Observations for Registration year
We see from above that minimum percentage is 0.002142 which is just 1 car in registration years: 1952, 1943 and few others.
The maximum is 6.76% - 2000, 6.29% - 2005, 6.21% - 1999

In [22]:
# Analyze the brand column
# Use loops to perform aggregation
print("Unique brands:\n" + str(autos['brand'].unique()))
print("Value counts:\n" + str(autos['brand'].value_counts()))
autos['brand'].describe()

Unique brands:
['peugeot' 'bmw' 'volkswagen' 'smart' 'ford' 'chrysler' 'seat' 'renault'
 'mercedes_benz' 'audi' 'sonstige_autos' 'opel' 'mazda' 'porsche' 'mini'
 'toyota' 'dacia' 'nissan' 'jeep' 'saab' 'volvo' 'mitsubishi' 'jaguar'
 'fiat' 'skoda' 'subaru' 'kia' 'citroen' 'chevrolet' 'hyundai' 'honda'
 'daewoo' 'suzuki' 'trabant' 'land_rover' 'alfa_romeo' 'lada' 'rover'
 'daihatsu' 'lancia']
Value counts:
volkswagen        9862
bmw               5137
opel              5022
mercedes_benz     4503
audi              4041
ford              3263
renault           2201
peugeot           1393
fiat              1197
seat               853
skoda              766
nissan             713
mazda              709
smart              661
citroen            654
toyota             593
hyundai            468
sonstige_autos     458
volvo              427
mini               409
mitsubishi         384
honda              366
kia                330
alfa_romeo         310
porsche            286
suzuki          

count          46681
unique            40
top       volkswagen
freq            9862
Name: brand, dtype: object

In [23]:
# Value counts of auto brands by normalizing to %
autos['brand'].value_counts(normalize=True).sort_values(ascending=False)*100

volkswagen        21.126368
bmw               11.004477
opel              10.758124
mercedes_benz      9.646323
audi               8.656627
ford               6.989996
renault            4.714980
peugeot            2.984083
fiat               2.564212
seat               1.827296
skoda              1.640925
nissan             1.527388
mazda              1.518819
smart              1.415994
citroen            1.400998
toyota             1.270324
hyundai            1.002549
sonstige_autos     0.981127
volvo              0.914719
mini               0.876159
mitsubishi         0.822604
honda              0.784045
kia                0.706926
alfa_romeo         0.664082
porsche            0.612669
suzuki             0.593389
chevrolet          0.569825
chrysler           0.351321
dacia              0.263490
daihatsu           0.250637
jeep               0.227073
subaru             0.214220
land_rover         0.209936
saab               0.164949
jaguar             0.156381
daewoo             0

In [24]:
# Select top 20 brands in list
selected_brands = autos['brand'].value_counts(normalize=True).sort_values(ascending=False).head(20)
print(selected_brands)
top20_brands = selected_brands.index
top20_brands

volkswagen        0.211264
bmw               0.110045
opel              0.107581
mercedes_benz     0.096463
audi              0.086566
ford              0.069900
renault           0.047150
peugeot           0.029841
fiat              0.025642
seat              0.018273
skoda             0.016409
nissan            0.015274
mazda             0.015188
smart             0.014160
citroen           0.014010
toyota            0.012703
hyundai           0.010025
sonstige_autos    0.009811
volvo             0.009147
mini              0.008762
Name: brand, dtype: float64


Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat', 'skoda', 'nissan', 'mazda', 'smart',
       'citroen', 'toyota', 'hyundai', 'sonstige_autos', 'volvo', 'mini'],
      dtype='object')

# Auto brands
We have 40 unique auto brands in the dataset. 3 top brands are listed and have > 10% of the total. 17 brands are above 1%, 23 brands are below 1%.

For our analysis we selected the top 20 brands as they appear to be more relevant.

In [25]:
# Dictionary of aggregate data for brands
brands_mean_prices = {}

for brand in top20_brands:
    current_brand = autos[autos["brand"] == brand]
    mean_price = current_brand["price"].mean()
    brands_mean_prices[brand] = int(mean_price)

brands_mean_prices

{'audi': 9336,
 'bmw': 8332,
 'citroen': 3779,
 'fiat': 2813,
 'ford': 3749,
 'hyundai': 5365,
 'mazda': 4112,
 'mercedes_benz': 8628,
 'mini': 10613,
 'nissan': 4743,
 'opel': 2975,
 'peugeot': 3094,
 'renault': 2474,
 'seat': 4397,
 'skoda': 6368,
 'smart': 3580,
 'sonstige_autos': 12338,
 'toyota': 5167,
 'volkswagen': 5402,
 'volvo': 4946}

# Top 20 brands mean prices
We see that Sonstige Autos has maximum mean price of 12338. The minimum is Renault: 2474. Audi, BMW, Mini, Mercedes Benz are the higher end prices, as the mean price is above 8000

# Exploring mileage


In [26]:
brands_mean_mileage = {}
# Use loop to calculate mean mileage for the brands
for brand in top20_brands:
    current_brand = autos[autos["brand"] == brand]
    mean_mileage = current_brand["odometer_km"].mean()
    brands_mean_mileage[brand] = int(mean_mileage)

# Convert both dictionaries to series objects
mean_mileage = pd.Series(brands_mean_mileage).sort_values(ascending=False)
mean_prices = pd.Series(brands_mean_prices).sort_values(ascending=False)
print(mean_mileage)
print(mean_prices)

volvo             138067
bmw               132572
mercedes_benz     130788
opel              129310
audi              129157
volkswagen        128707
renault           128071
peugeot           127153
mazda             124464
ford              124266
seat              121131
citroen           119694
nissan            118330
fiat              117121
toyota            115944
skoda             110848
hyundai           106442
smart              99326
sonstige_autos     89956
mini               88105
dtype: int64
sonstige_autos    12338
mini              10613
audi               9336
mercedes_benz      8628
bmw                8332
skoda              6368
volkswagen         5402
hyundai            5365
toyota             5167
volvo              4946
nissan             4743
seat               4397
mazda              4112
citroen            3779
ford               3749
smart              3580
peugeot            3094
opel               2975
fiat               2813
renault            2474
dtype: 

In [27]:
topbrandsDF = pd.DataFrame(mean_mileage, columns = ['mean_mileage'])
topbrandsDF['mean_price'] = mean_prices
topbrandsDF

mean_mileage  mean_price
volvo                 138067        4946
bmw                   132572        8332
mercedes_benz         130788        8628
opel                  129310        2975
audi                  129157        9336
volkswagen            128707        5402
renault               128071        2474
peugeot               127153        3094
mazda                 124464        4112
ford                  124266        3749
seat                  121131        4397
citroen               119694        3779
nissan                118330        4743
fiat                  117121        2813
toyota                115944        5167
skoda                 110848        6368
hyundai               106442        5365
smart                  99326        3580
sonstige_autos         89956       12338
mini                   88105       10613

# Top 20 brands mileage and price analysis
From above dataframe, we see there is a slight increase in price as mileage is going down. But it is not consistent, there are 3 brands in top 5 that have higher price - BMW, Mercedes and Audi. 

In [28]:
autos.head(30)

dateCrawled                                               name  \
0   2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1   2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2   2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3   2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4   2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5   2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6   2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7   2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8   2016-03-22 16:51:34                                         Seat_Arosa   
9   2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
11  2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12  2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13  2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14  2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15  2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16  2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17  2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18  2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19  2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20  2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21  2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22  2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23  2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24  2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25  2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26  2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
28  2016-03-19 21:56:19                                      MINI_Cooper_D   
29  2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
30  2016-03-14 11:47:31                         Peugeot_206_Unfallfahrzeug   
31  2016-03-14 16:53:09                          Noch_gut_erhaltenen_C_320   

    seller offerType  price   abtest vehicleType  registration_year  \
0   privat   Angebot   5000  control         bus               2004   
1   privat   Angebot   8500  control   limousine               1997   
2   privat   Angebot   8990     test   limousine               2009   
3   privat   Angebot   4350  control  kleinwagen               2007   
4   privat   Angebot   1350     test       kombi               2003   
5   privat   Angebot   7900     test         bus               2006   
6   privat   Angebot    300     test   limousine               1995   
7   privat   Angebot   1990  control   limousine               1998   
8   privat   Angebot    250     test         NaN               2000   
9   privat   Angebot    590  control         bus               1997   
11  privat   Angebot    350  control         NaN               2000   
12  privat   Angebot   5299  control  kleinwagen               2010   
13  privat   Angebot   1350  control   limousine               1999   
14  privat   Angebot   3999     test  kleinwagen               2007   
15  privat   Angebot  18900     test       coupe               1982   
16  privat   Angebot    350     test       kombi               1999   
17  privat   Angebot   5500     test       coupe               1990   
18  privat   Angebot    300  control         bus               1995   
19  privat   Angebot   4150  control         suv               2004   
20  privat   Angebot   3500     test       kombi               2003   
21  privat   Angebot  41500     test      cabrio               2004   
22  privat   Angebot  25450  co

In [29]:
# Find most common brand / model combinations
brandModelDF = autos[['brand','model']]
brandModelDF.head()

brand   model
0     peugeot  andere
1         bmw     7er
2  volkswagen    golf
3       smart  fortwo
4        ford   focus

In [30]:
# Top 10 brands
top10brands = brandModelDF['brand'].value_counts().sort_values(ascending=False).head(10).index
top10brands

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford', 'renault',
       'peugeot', 'fiat', 'seat'],
      dtype='object')

In [31]:
# Get a new dataframe of top 10 brands models
top10brandsDF = pd.DataFrame(columns = ['brand','model'])
#print(topbrandsrowsDF)
for current_brand in top10brands:
    currentbrandDF = brandModelDF[brandModelDF['brand'] == current_brand]
    #print(currentbrandDF)
    top10brandsDF = top10brandsDF.append(currentbrandDF,ignore_index=True)
#topbrandsDF = brandModelDF[brandModelDF['brand']==current_brand
top10brandsDF.tail()

brand   model
37467  seat   arosa
37468  seat  toledo
37469  seat   ibiza
37470  seat   arosa
37471  seat   ibiza

In [32]:
# For the selected top 10 brands, get counts
top10brandsDF.groupby('brand').count()
top10brandsDF.groupby('model')['brand'].count().sort_values(ascending=False)
#top10brandsDF.groupby(['brand'])['model'].value_counts()

#top10brandsDF.groupby(['brand','model']).count()

model
golf           3707
3er            2615
andere         1642
polo           1609
corsa          1592
passat         1349
astra          1348
a4             1231
c_klasse       1136
5er            1132
e_klasse        958
a3              825
a6              797
focus           762
fiesta          722
transporter     674
twingo          615
2_reihe         600
vectra          544
a_klasse        539
1er             521
mondeo          479
clio            473
touran          433
punto           415
zafira          394
ka              349
megane          335
ibiza           328
3_reihe         322
               ... 
doblo            41
kuga             41
alhambra         41
q7               40
altea            40
toledo           40
jetta            38
cl               38
fusion           33
phaeton          31
glk              30
6er              30
modus            29
q3               28
v_klasse         24
g_klasse         24
calibra          22
antara           20
cc            

In [33]:
# For the selected top 10 brands, get counts of model grouped by brand and sort descending
top10brandsmodelgrouped = top10brandsDF.groupby(['brand', 'model']).size().sort_values(ascending=False)
top10brandsmodelgrouped = top10brandsmodelgrouped.reset_index()
top10brandsmodelgrouped.columns=np.where(top10brandsmodelgrouped.columns==0, 'count', top10brandsmodelgrouped.columns) #replace the default 0 to 'count'
print(top10brandsmodelgrouped)

             brand        model  count
0       volkswagen         golf   3707
1              bmw          3er   2615
2       volkswagen         polo   1609
3             opel        corsa   1592
4       volkswagen       passat   1349
5             opel        astra   1348
6             audi           a4   1231
7    mercedes_benz     c_klasse   1136
8              bmw          5er   1132
9    mercedes_benz     e_klasse    958
10            audi           a3    825
11            audi           a6    797
12            ford        focus    762
13            ford       fiesta    722
14      volkswagen  transporter    674
15         renault       twingo    615
16         peugeot      2_reihe    600
17            opel       vectra    544
18   mercedes_benz     a_klasse    539
19             bmw          1er    521
20            ford       mondeo    479
21         renault         clio    473
22   mercedes_benz       andere    439
23      volkswagen       touran    433
24            fiat       

# Most common brand/model combinations
We can see below the top 5 brand/model combinations with the number available.
- volkswagen golf 3707
- bmw 3er 2615
- volkswagen polo 1609
- opel corsa 1592
- volkswagen passat 1349

In [34]:
# Split odometer_km into groups 
autos['odometer_km'].value_counts()

150000    30085
125000     4857
100000     2058
90000      1673
80000      1375
70000      1187
60000      1128
50000       993
40000       797
5000        785
30000       760
20000       742
10000       241
Name: odometer_km, dtype: int64

In [37]:
# Break into bins
bins = [0,40000,70000,100000,130000,160000]

# Create the names for the bins.
# Group names will have 1 less element in list than the bins defined
group_names = ["<40K","40K-70K","70K-100K","100K-130K","130K+"]

# For purchase_data, a new column of Age Group is added
autos["odometer_group"] = pd.cut(autos["odometer_km"],bins,labels=group_names)
autos.head()


dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType  price   abtest vehicleType  registration_year    gearbox  \
0  privat   Angebot   5000  control         bus               2004    manuell   
1  privat   Angebot   8500  control   limousine               1997  automatik   
2  privat   Angebot   8990     test   limousine               2009    manuell   
3  privat   Angebot   4350  control  kleinwagen               2007  automatik   
4  privat   Angebot   1350     test       kombi               2003    manuell   

   powerPS      ...       odometer_km  registration_month  fuelType  \
0      158      ...            150000                   3       lpg   
1      286      ...            150000                   6    benzin   
2      102      ...             70000                   7    benzin   
3       71      ...             70000                   6    benzin   
4        0      ...            150000                   7    benzin   

        brand unrepaired_damage           ad_created nrOfPictures  postalCode  \
0     peugeot              nein  2016-03-26 00:00:00            0       79588   
1         bmw              nein  2016-04-04 00:00:00            0       71034   
2  volkswagen              nein  2016-03-26 00:00:00            0       35394   
3       smart              nein  2016-03-12 00:00:00            0       33729   
4        ford              nein  2016-04-01 00:00:00            0       39218   

              lastSeen odometer_group  
0  2016-04-06 06:45:54          130K+  
1  2016-04-06 14:45:08          130K+  
2  2016-04-06 20:15:37        40K-70K  
3  2016-03-15 03:16:28        40K-70K  
4  2016-04-01 14:38:50          130K+  

[5 rows x 21 columns]

In [38]:
# Use aggregation to see if average prices follow any patterns based on mileage
autos.groupby('odometer_group')['price'].mean().sort_values(ascending=False)

odometer_group
<40K         15212.965714
40K-70K      12290.308041
70K-100K      8669.556600
100K-130K     6214.022030
130K+         3767.927107
Name: price, dtype: float64

# Analysis of price compared to odometer group
It is seen that as the odometer mileage increases, price decreases and they are inversely proportional.

For Odomter reading < 40K, mean price is 15213. 70K-100K has 8670, 130K+ has 3768.

In [40]:
# How much cheaper are cars with damage than their non-damaged counterparts?
print("Mean price is:" + str(autos['price'].mean()))
autos.groupby('unrepaired_damage')['price'].mean().sort_values(ascending=False)

Mean price is:5977.716801268182


unrepaired_damage
nein    7164.033103
ja      2241.146035
Name: price, dtype: float64

# Car damage related to price
There is close to 5000 dollars price difference between cars with unrepaired damage and those that have had repairs on damages / no damage.
Unrepaired average price is: 2241 
Repaired average price is: 7164
The average price for all cars is 5978.